src/tod_manager.cpp

```diff
@@ -255,8 +255,7 @@ const time_of_day tod_manager::get_illuminated_time_of_day(const unit_map & unit
 			    !itor->incapacitated())
 			{
 				unit_ability_list illum = itor->get_abilities("illuminates");
-				unit_abilities::effect illum_effect(illum, terrain_light, false);
-				const int unit_mod = illum_effect.get_composite_value();
+				const int unit_mod = illum.highest("value").first;
```

Comment:
using highest ("value") for illuminates is sufficient and allows to redefine the calculation rules of get_composite_value proposed by @gfgtdf without alterates that in a bad ways.

Reply:
maybe you could make another seperate pr withotu this line ? Just so that i can test the mentioned bug myself.

Reply:
another pr without tod manager, ok.